Can't install for React 18 without legacy dependency resolution

peerDependencies of Blade need to be updated to support React 18.

https://github.com/razorpay/blade/blob/master/packages/blade/package.json#L171-L172

Currently trying to install Blade in a project with React v18 results in an error:

```sh
npm i @razorpay/blade
```
> npm ERR! Could not resolve dependency:
npm ERR! peer react@"^17.0.2" from @razorpay/blade@5.5.0
npm ERR! node_modules/@razorpay/blade
npm ERR!   @razorpay/blade@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

## Software versions
Node: v18.13.0
npm: 8.19.3

--- a/.changeset/wet-hotels-doubt.md
+++ b/.changeset/wet-hotels-doubt.md
@@ -1,0 +1,5 @@
+---
+'@razorpay/blade': patch
+---
+
+fix(blade): update peerDependencies to support react v18

--- a/packages/blade/package.json
+++ b/packages/blade/package.json
@@ -168,8 +168,8 @@
     "tsconfig-paths-webpack-plugin": "3.5.2"
   },
   "peerDependencies": {
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
+    "react": ">=17",
+    "react-dom": ">=17",
     "react-native": "^0.66.5",
     "react-native-reanimated": "^2.6.0",
     "react-native-svg": "^12.3.0",

